companion/lib/main.ts

```diff
@@ -73,13 +92,13 @@ program.command('start', { isDefault: true, hidden: true }).action(() => {
 
 	if (options.syslogEnable) {
 		const opt: SyslogTransportOptions = {}
-		if (options.syslogHost) opt.host = options.syslogHost
+		if (isValidIPv4(options.syslogHost)) opt.host = options.syslogHost
 		if (options.syslogPort) {
 			const port = Number.parseInt(options.syslogPort)
 			if (!Number.isNaN(port) && port > 100 && port <= 65535) opt.port = port
 		}
 		if (options.syslogTcp) opt.protocol = 'tcp4'
-		if (options.syslogLocalhost) opt.localhost = options.syslogLocalhost
+		opt.localhost = options.syslogLocalhost ? options.syslogLocalhost : os.hostname()
```

Comment:
```suggestion
		opt.localhost = options.syslogLocalhost?.trim() || os.hostname()
```

I suspect trimming the value would be a good idea, partly to ensure it doesnt get set to just something like ` ` and result in not sending anything.

Feel free to ignore if you think it should remain as you have written